UtilitiesProject/Helpers.cs

```diff
@@ -1,10 +1,199 @@
-﻿namespace UtilitiesProject
+﻿using Microsoft.AspNetCore.Mvc.Rendering;
+using ModelsProject;
+using Newtonsoft.Json.Linq;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Xml.Linq;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+[assembly: InternalsVisibleTo("MVCDemoApp")]
+namespace UtilitiesProject
 {
-	public class Helpers
+    public static class Helpers
     {
+        #region General Methods or Functions.
+
+        internal static List<string> GetColumnNames<T>(List<T> models)
+        {
+            if (models == null || !models.Any())
+            {
+                return new List<string>();
+            }
+
+            // Get the type of the model
+            Type modelType = typeof(T);
+
+            // Get all public properties of the model
+            PropertyInfo[] properties = modelType.GetProperties(BindingFlags.Public | BindingFlags.Instance);
+
+            // Extract property names
+            return properties.Select(p => p.Name).ToList();
+        }
+
+        internal static List<string> ParseJArray(string json)
+        {
+            List<string> result = new List<string>();
+            JArray array = JArray.Parse(json);
+
+            foreach (JObject content in array.Children<JObject>())
+            {
+                List<string> keys = content.Properties().Select(p => p.Name).ToList();
+                result.AddRange(keys);
+            }
+
+            return result;
+        }
+
+        #endregion
+
+        #region Apply Filter on Datagrid
+
+        /// <summary>
+        /// Apply Filter on Datagrid
+        /// </summary>
+        /// <typeparam name="T">Model</typeparam>
+        /// <param name="query">List of data with query way</param>
+        /// <param name="filter">Data Grid with filter</param>
+        /// <returns></returns>
+        internal static IQueryable<T> ApplyFilter<T>(IQueryable<T> query, object filter)
+        {
+            IQueryable<T> result;
+            try
+            {
+                int propertyNamePosition = 0, operationPosition = 1, filterPosition = 2;
+                if (filter is IList<object> filterList)
+                {
+                    //it is checking all way.
+                    if (filterList[propertyNamePosition].GetType().Name == typeof(string).Name)
+                    {
+                        //single filter
+                        query = SingleLogic(query, filterList, propertyNamePosition, operationPosition, filterPosition).AsQueryable();
```

Comment:
I think we don't need "AsQueryable" here as you are already returning IQueryable

Reply:
I have collection of list. So, When I called **ApplyFilter** that time has Iqueryable. That time I need to convert list to IQuarable.
Anyway, let me check again to remove it and test again.